pkg/controller/core/workload_controller.go

```diff
@@ -137,16 +137,18 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	// If the workload has rejected admission checks and is not admitted mark it as finished.
-	// If it has rejected admission checks and is admitted, continue the reconcile in  order to trigger its eviction.
-	if rejectedChecks := workload.GetRejectedChecks(&wl); len(rejectedChecks) > 0 && !workload.IsAdmitted(&wl) {
-		log.V(3).Info("Workload has Rejected admission checks, Finish with failure")
-		err := workload.UpdateStatus(ctx, r.client, &wl, kueue.WorkloadFinished,
-			metav1.ConditionTrue,
-			"AdmissionChecksRejected",
-			fmt.Sprintf("Admission checks %v are rejected", rejectedChecks),
-			constants.KueueName)
-		return ctrl.Result{}, err
+	// If the workload is not admitted or in the eviction process.
+	if !workload.IsAdmitted(&wl) && (!apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) || !workload.HasQuotaReservation(&wl)) {
```

Comment:
```suggestion
	if !(workload.IsAdmitted(&wl) || (apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) && workload.HasQuotaReservation(&wl))) {
```
Seems more readable to have a single negation, but no strong view.

Reply:
Maybe a better alternative would be to move this logic about "Finalize on Reject" after we have handled evictions.